src/sql/orderbook/batch_rewards.sql

```diff
@@ -0,0 +1,40 @@
+with hashed_observations as (SELECT block_number,
+                                    tx_hash,
+                                    effective_gas_price * gas_used as execution_cost,
+                                    surplus,
+                                    fee,
+                                    auction_id
+                             FROM settlement_observations so
+                                    LEFT OUTER JOIN LATERAL (
+                               SELECT tx_hash, solver, tx_nonce, tx_from
+                               FROM settlements s
+                               WHERE s.block_number = so.block_number
+                                 AND s.log_index > so.log_index
+                               ORDER BY s.log_index
+                               LIMIT 1
```

Comment:
I'm just wondering: Are you trying to find all settlement_observations that have a matching row in the settlement table?

Reply:
I'm unsure how much performance dependent we are on this query, but using a `WHERE EXISTS` might be better for performance than a join.